server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/websockets/CBEventsLongPollingServlet.java

```diff
@@ -258,4 +251,78 @@ public static void sendError(
     }
 
 
+    /**
+     * Resolves a {@link BaseWebSession} from an incoming HTTP request.
+     * <p>
+     * Implementations may use different resolution strategies
+     * (e.g. cookies, headers, tokens).
+     * </p>
+     *
+     */
+    public interface WebSessionResolver {
+        /**
+         * @return session or null if resolver is not applicable
+         */
+        @Nullable
+        BaseWebSession resolve(@NotNull HttpServletRequest req);
+    }
+
+    public static class CookieWebSessionResolver implements WebSessionResolver {
+
+        @Nullable
+        @Override
+        public BaseWebSession resolve(@NotNull HttpServletRequest req) {
+
+            String cookieSid = ServletAppUtils.getRequestCookie(req, CBConstants.CB_SESSION_COOKIE_NAME);
+
+            if (cookieSid == null) {
+                return null;
+            }
+
+            WebHttpRequestInfo info = new WebHttpRequestInfo(
```

Comment:
Move it to createRequestInfo. We have pretty same code 3 times here.
Also we cannot use getRemoteAddr directly, see how we do it in https://github.com/dbeaver/pro/issues/6123